routers/web/repo/wiki.go

```diff
@@ -276,7 +276,7 @@ func renderViewPage(ctx *context.Context) (*git.Repository, *git.TreeEntry) {
 			return nil, nil
 		}
 		ctx.Data["footerPresent"] = footerContent != nil
-		ctx.Data["footerEscapeStatus"], ctx.Data["footerContent"] = charset.EscapeControlString(buf.String())
+		ctx.Data["footerEscapeStatus"], ctx.Data["footerContent"] = charset.EscapeControlHTML(buf.String(), ctx.Locale, 0xa0)
```

Comment:
Instead of always passing `0xa0`, we should probably define a constant to use for these calls.

Reply:
+1

Or just accept all 'NO-BREAK SPACE'? Or have some comments about why sometimes the nbsp is acceptable while sometimes not. (it's not clear to me why. not sure whether other readers know the real reason).

Reply:
In rendered forms `&nbsp;` should generally not be considered as an invisible character.

In source code `&nbsp;` is likely to be ambiguous and confusing.

I regret adding it in this PR and should have left it for another PR.

Reply:
I've made a common render func for this and added a comment.